WultraMobileTokenSDK/Oidc/Utils/WMTOidcUtils.swift

```diff
@@ -0,0 +1,123 @@
+//
+// Copyright 2025 Wultra s.r.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions
+// and limitations under the License.
+//
+
+import Foundation
+
+/// Utility class for OIDC
+public class WMTOidcUtils {
+    
+    /// Creates PKCE codes, returning a Result wrapping the codes.
+    public static func createPKCE(dataLength: Int) throws -> WMTPKCECodes {
+        let minLength: Int = 32 // min is 32-octet sequence == Base64 43 URL safe characters
+        let maxLength: Int = 96 // max is 96-octet sequence == Base64 128 URL safe characters
+        let length = (dataLength > minLength && dataLength < maxLength) ? dataLength : minLength
+        
+        let codeVerifier = try getRandomBase64UrlSafe(dataLength: length)
+        let codeChallenge = try generateCodeChallenge(verifier: codeVerifier)
+        return WMTPKCECodes(codeVerifier: codeVerifier, codeChallenge: codeChallenge)
+    }
+
+    /// Generates a random Base64 URL-safe string of the given length.
+    public static func getRandomBase64UrlSafe(dataLength: Int) throws -> String {
+        var randomBytes = [Int8](repeating: 0, count: dataLength)
+        
+        // Fill bytes with secure random data
+        let status = SecRandomCopyBytes(kSecRandomDefault, dataLength, &randomBytes)
+        
+        // A status of errSecSuccess indicates success
+        guard status == errSecSuccess else { throw WMTError(reason: .randomBytesFailed) }
+        
+        // Convert bytes to Data
+        let data = Data(bytes: randomBytes, count: dataLength)
+        return data.base64EncodedString().safeUrlString
+    }
+    
+    /// Creates an authorization URL.
```

Comment:
The comment could be longer. Some exaple usage could be eg:
```
/// Creates an authorization URL.
///
/// URL is then used to open browser to go through the user login
```

or something like this.